### ISSUE TITLE
Unable to find step def. for step containing parentheses

### ISSUE THREAD

**Issue by emily25inGoFreight** [alexkrechik/VSCucumberAutoComplete#405]
**Describe the bug**
Unable to find step def. for step containing parentheses, e.g., When the user click 'Gross Weight (SHPR)' for box(2).
For situation like parentheses in braces, it's fine. But when there are only parentheses, then it won't work.
Whether there are quotes does not affect the result. 

**To Reproduce**
1. Go to a feature file
2. Right click on the step in editor, and click 'Go to Definition'
3. See error

**Expected behavior**
Expected to find the step definition.

**Cucumberautocomplete part of VSCode settings:**
```
{
    "cucumberautocomplete.steps": [
        "features/steps/*.py"
    ],
    "cucumberautocomplete.syncfeatures": "features/*.feature",
    "cucumberautocomplete.strictGherkinCompletion": true,
    "cucumberautocomplete.skipDocStringsFormat": true
}
```

**Step definition:**
If applicable, add example of step definition:
```python
@When("the user click 'Gross Weight (SHPR)' for box({box_index})")
def step_impl(context, box_index):
    pass
```

**Gherkin step line**
If applicable, add step line from gherkin file
`When the user click 'Gross Weight (SHPR)' for box(2)`

**Comment by koichirok** [comment]
[Parentheses are used for optional text in cucumber expressions](https://cucumber.io/docs/cucumber/cucumber-expressions/#optional-text), and must be escaped in order for the parentheses to be matched literally.

So example step definition above should be:

```
@When("the user click 'Gross Weight \(SHPR)' for box\({box_index})")
def step_impl(context, box_index):
    pass
```

However, even if the step definition is correct, the result is still the same. Cannot jump to the step definition with 'Go to Definition'.

As a workaround, I use regular expressions for step definitions in such cases.

**Comment by ahmdgawad** [comment]
I have similar issue in Java:
`@When("a file named {string} is created")`
Was unable to find step for "When a file named 'abc' is created.

@koichirok Can you please mention how you used regex in step definitions?

**Comment by emily25inGoFreight** [comment]
> I have similar issue in Java:
> `@When("a file named {string} is created")`
> Was unable to find step for "When a file named 'abc' is created.
> 
> @koichirok Can you please mention how you used regex in step definitions?

In this case, maybe you should add single quote aound `{string}`, like `@When("a file named '{string}' is created")`

**Comment by emily25inGoFreight** [comment]
> [Parentheses are used for optional text in cucumber expressions](https://cucumber.io/docs/cucumber/cucumber-expressions/#optional-text), and must be escaped in order for the parentheses to be matched literally.
> 
> So example step definition above should be:
> 
> ```
> @When("the user click 'Gross Weight \(SHPR)' for box\({box_index})")
> def step_impl(context, box_index):
>     pass
> ```
> 
> However, even if the step definition is correct, the result is still the same. Cannot jump to the step definition with 'Go to Definition'.
> 
> As a workaround, I use regular expressions for step definitions in such cases.

Since changing all test case to regular expressions may require lots of effort, my current approach is add this in settings
```
"cucumberautocomplete.customParameters": [
        {
            "parameter":"(SHPR)",
            "value":"\\(SHPR\\)"
        }
    ]
But I'm still hoping that there could be an update for this.
```

**Comment by ahmdgawad** [comment]
> > [Parentheses are used for optional text in cucumber expressions](https://cucumber.io/docs/cucumber/cucumber-expressions/#optional-text), and must be escaped in order for the parentheses to be matched literally.
> > So example step definition above should be:
> > ```
> > @When("the user click 'Gross Weight \(SHPR)' for box\({box_index})")
> > def step_impl(context, box_index):
> >     pass
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > However, even if the step definition is correct, the result is still the same. Cannot jump to the step definition with 'Go to Definition'.
> > As a workaround, I use regular expressions for step definitions in such cases.
> 
> Since changing all test case to regular expressions may require lots of effort, my current approach is add this in settings
> 
> ```
> "cucumberautocomplete.customParameters": [
>         {
>             "parameter":"(SHPR)",
>             "value":"\\(SHPR\\)"
>         }
>     ]
> ```
> 
> But I'm still hoping that there could be an update for this.

Thanks, it worked that way, However I made a long list as the plugin was catching braces in javadoc also. And its nearly impossible to change all test cases (1000+).
Hope to get a fix for this.

**Comment by alexkrechik** [comment]
Like the same with https://github.com/alexkrechik/VSCucumberAutoComplete/issues/222

**Comment by alexkrechik** [comment]
Should be fixed with `"cucumberautocomplete.pureTextSteps": true` option using after 2.16.0 release.

**Comment by alexkrechik** [comment]
This issue is to be fixed or implemented in the 3.0.0 version of the extension. If this issue is unresolved or there are additional notes, please create a new ticket with the necessary details.
